mm2src/coins/eth/v2_activation.rs

```diff
@@ -620,14 +621,15 @@ pub async fn eth_coin_from_conf_and_request_v2(
     )
     .await?;
 
-    let chain_id = conf["chain_id"].as_u64().ok_or(EthActivationV2Error::ChainIdNotSet)?;
     let web3_instances = match (req.rpc_mode, &priv_key_policy) {
         (EthRpcMode::Default, EthPrivKeyPolicy::Iguana(_) | EthPrivKeyPolicy::HDWallet { .. })
         | (EthRpcMode::Default, EthPrivKeyPolicy::Trezor) => {
             build_web3_instances(ctx, ticker.to_string(), req.nodes.clone()).await?
         },
         #[cfg(target_arch = "wasm32")]
         (EthRpcMode::Metamask, EthPrivKeyPolicy::Metamask(_)) => {
+            // Metamask doesn't support native Tron
+            let chain_id = chain_spec.chain_id().ok_or(EthActivationV2Error::ChainIdNotSet)?;
```

Comment:
Shouldn't we return an explicit error indicating chain not supported for Metamask?

Reply:
Makes sense, will fix it.